platform/mellanox/mlnx-platform-api/sonic_platform/component.py

```diff
@@ -182,8 +184,12 @@ def __unstage_update(self, image_path):
         except subprocess.CalledProcessError as e:
             raise RuntimeError("Failed to unstage firmware update: {}".format(str(e)))
 
-    def __trigger_update(self):
-        cmd = self.ONIE_FW_UPDATE_CMD_UPDATE
+    def __trigger_update(self, reboot):
+
+        if reboot:
+            cmd = self.ONIE_FW_UPDATE_CMD_UPDATE
+        else:
+            cmd = self.ONIE_FW_UPDATE_CMD_INSTALL
```

Comment:
@alexrallen FYI: in case a several FW binaries are staged, ONiE FW Framework can update only one at time. So eventually it will require a few cold reboots

Reply:
Understood.